fuzztest/domain_core.h

```diff
@@ -116,8 +116,8 @@ class DomainBuilder {
           iter.second != nullptr && iter.second->has_value(),
           "Some domain is not set yet!");
     }
-    return OwningDomain<T>(GetIndirect<T>(name)->template GetAs<Domain<T>>(),
```

Comment:
It leads to UB, if second argument evaluated first, since GetIndirect uses `domain_lookup_table_` inside